### PR TITLE
Setup prometheus again without minikube

### PR DESCRIPTION
**Pull request by SagaRut**
This is the basic to the prometheus setup. Had to refactor everything to not use a different namespace for monitoring because prometheus apparently can't monitor unless its in the same namespace...tried some fixes for it but none worked.

### CODE REVIEW

**Review by Timdnb — APPROVED** [review]

**Comment by Timdnb** [comment]
Only thing maybe is that also here the .yml file are applied on the controller before the other nodes existing, so the cluster does not really have an opportunity to distribute them (if that even would make sense)

**Comment by SagaRut** [comment]
> Only thing maybe is that also here the .yml file are applied on the controller before the other nodes existing, so the cluster does not really have an opportunity to distribute them (if that even would make sense)

Ah yeah I know, but I think this can all be on the controller, there is a question about this on mattermost, what should be where but no reply yet

**Review by Guthax — APPROVED** [review]

File: README.md
Comment (Timdnb) [inline README.md:1]: with the current setup, it is nicer if you just do everything from host, so

export KUBECONFIG=./playbooks/k3s.yaml
kubectl get services

that should work the same, but no need to ssh into the controller. Same goes for the grafana one, so maybe you could update that one too in this PR?

File: playbooks/setup_k8s_cluster.yml
Comment (Timdnb) [inline playbooks/setup_k8s_cluster.yml:1]: For me when I run it, it fails at "Add Helm repositories" with:

fatal: [controller]: FAILED! => {"changed": true, "cmd": "export KUBECONFIG=/vagrant/playbooks/k3s.yaml\nhelm repo add prom-repo https://prometheus-community.github.io/helm-charts\nhelm repo add kubernetes-dashboard https://kubernetes.github.io/dashboard/\nhelm repo update\n", "delta": "0:00:00.001752", "end": "2024-06-07 11:45:26.112964", "msg": "non-zero return code", "rc": 127, "start": "2024-06-07 11:45:26.111212", "stderr": "/bin/sh: 2: helm: not found\n/bin/sh: 3: helm: not found\n/bin/sh: 4: helm: not found", "stderr_lines": ["/bin/sh: 2: helm: not found", "/bin/sh: 3: helm: not found", "/bin/sh: 4: helm: not found"], "stdout": "", "stdout_lines": []}

However you didnt change anything for that section, so I am not sure why I am getting this. For you everything runs as expected Im assuming right?

Comment (Timdnb) [reply playbooks/setup_k8s_cluster.yml:1]: Okay just ran the main branch again, and get the same error there even though that worked previously (I've seen the grafana dashboard before haha). So I am guessing this is something on my side, even though I have no clue what

File: README.md
Comment (Timdnb) [reply README.md:1]: What error do you get? And I assume you have kubectl installed on your host machine too right?

Comment (SagaRut) [reply README.md:1]: Updated the readme

**Review by Timdnb — APPROVED** [review]

**Pull request merged**